.github/workflows/publish_pypi.yml

```diff
@@ -2,251 +2,38 @@ name: Build and publish package to PyPI
-          CIBW_ENVIRONMENT: >
-            PYBAMM_DISABLE_TELEMETRY="true"
-          # 10.13 for Intel (macos-13), 11.0 for Apple Silicon (macos-14 and macos-latest)
-          MACOSX_DEPLOYMENT_TARGET: ${{ matrix.os == 'macos-14' && '11.0' || '10.13' }}
-          CIBW_ARCHS_MACOS: auto
-          CIBW_BEFORE_BUILD: python -m pip install setuptools delocate
-          CIBW_REPAIR_WHEEL_COMMAND: |
-            if [[ $(uname -m) == "x86_64" ]]; then
-              delocate-listdeps {wheel} && delocate-wheel -v -w {dest_dir} {wheel}
-            elif [[ $(uname -m) == "arm64" ]]; then
-              # Use higher macOS target for now since casadi/libc++.1.0.dylib is still not fixed
-              delocate-listdeps {wheel} && delocate-wheel -v -w {dest_dir} {wheel} --require-target-macos-version 11.1
-              for file in {dest_dir}/*.whl; do mv "$file" "${file//macosx_11_1/macosx_11_0}"; done
-            fi
-          CIBW_TEST_EXTRAS: "all,dev,jax"
-          CIBW_TEST_COMMAND: |
-            set -e -x
-            python -m pytest -m cibw {project}/tests/unit
-
-      - name: Upload wheels for macOS (amd64, arm64)
-        uses: actions/upload-artifact@65c4c4a1ddee5b72f698fdd19549f0f0fb45cf08 # v4.6.0
-        with:
-          name: wheels_${{ matrix.os }}
-          path: ./wheelhouse/*.whl
-          if-no-files-found: error
-
-  build_sdist:
-    name: Build SDist
+  Build:
     runs-on: ubuntu-latest
 
     steps:
+      - name: Set up Python
       - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
       - uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
         with:
           python-version: 3.12
 
-      - name: Build SDist
-        run: pipx run build --sdist
+      - name: Build wheel
+        run: |
+          pipx run build --outdir deploy
 
-      - name: Upload SDist
+      - name: Upload package
         uses: actions/upload-artifact@65c4c4a1ddee5b72f698fdd19549f0f0fb45cf08 # v4.6.0
         with:
-          name: sdist
-          path: ./dist/*.tar.gz
+          name: wheels
+          path: deploy/
           if-no-files-found: error
 
-  publish_pypi:
-    # This job is only of value to PyBaMM and would always be skipped in forks
-    if: github.event_name != 'schedule' && github.repository == 'pybamm-team/PyBaMM'
-    name: Upload package to PyPI
-    needs: [
-      build_manylinux_wheels,
-      build_macos_wheels,
-      build_windows_wheels,
-      build_sdist
-    ]
+  Publish:
+    needs: Build
     runs-on: ubuntu-latest
-    environment:
-      name: pypi
-      url: https://pypi.org/p/pybamm
+    environment: pypi
     permissions:
       id-token: write
 
     steps:
-      - name: Download all artifacts
-        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
-        with:
-          path: artifacts
-          merge-multiple: true
-
-      - name: Sanity check downloaded artifacts
-        run: ls -lA artifacts/
```

Comment:
I would prefer keeping this too

Reply:
It does not seem like this really does any good. The build step prints them, and if this returns nothing it would not cause the job to stop